src/test/java/com/segment/analytics/android/integration/intercom/IntercomTest.java

```diff
@@ -148,7 +149,20 @@ public void identifyWithCompany() {
         company.put("name", "Acme");
         traits.put("company", company);
         integration.identify(new IdentifyPayloadBuilder().traits(traits).build());
-        verify(intercom).updateUser(any(UserAttributes.class));
+
+        Company expectedCompany = new Company.Builder()
+            .withCompanyId("456")
+            .withName("Acme")
+            .build();
+
+        UserAttributes expectedUserAttributes = new UserAttributes.Builder()
+            .withCompany(expectedCompany)
+            .build();
+
+        ArgumentCaptor<UserAttributes> attributesArgumentCaptor = ArgumentCaptor.forClass(UserAttributes.class);
+        verify(intercom).updateUser(attributesArgumentCaptor.capture());
+
+        assertThat(expectedUserAttributes).isEqualToComparingFieldByFieldRecursively(attributesArgumentCaptor.getValue());
```

Comment:
Does this need `isEqualToComparingFieldByFieldRecursively`?

Reply:
I'm not sure what a better alternative would be. UserAttributes doesn't include any getter methods, and we want to ensure that UserAttribute values match in each object instance. Let me know if there's something else that may work better here.

Reply:
Is `.isEqualTo` not sufficient?

Reply:
Hmmm, the test fails with `.isEqualTo` - the error indicates that new UserAttributes() =! new UserAttributes(), but definitely passes with `isEqualToComparingFieldByFieldRecursively` - let me know if I'm missing something really obvious. Thanks!

Reply:
Ok, this is good for now then.